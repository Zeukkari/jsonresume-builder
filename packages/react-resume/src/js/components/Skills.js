import React from 'react'

import { Box } from 'grommet'

import { SectionTitle, TagSection, ResumeSection } from './common'

const Skills = ({ skills }) => {
  const renderItems = skills.map(item => {
    return (
      <Box
        key={item.name}
        border={{
          side: 'all',
          color: 'red',
          size: 'xsmall',
          style: 'dashed',
        }}
        pad={{ bottom: 'medium' }}
      >
        <Box>{item.name}</Box>
        <Box>
          <Box direction="row">{item.level}</Box>
          <TagSection tags={item.keywords}>Keywords</TagSection>
        </Box>
      </Box>
    )
  })

  return (
    <Box margin="none" pad="none" border="all">
      <ResumeSection>
        <SectionTitle>Skills</SectionTitle>
        <Box direction="column">{renderItems}</Box>
      </ResumeSection>
    </Box>
  )
}

Skills.defaultProps = {
  skills: [],
}

export { Skills }
