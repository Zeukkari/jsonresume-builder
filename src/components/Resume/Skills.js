import React from 'react'

import { Box, Heading, Text } from 'grommet'

import { ResumeSection } from './common'

const Skills = ({ skills }) => {
  const renderItems = skills.map(item => {
    return (
      <Box key="item.name">
        <Heading>{item.name}</Heading>
        <Heading>{item.level}</Heading>

        <Text>Highlights</Text>
        <Box direction="column">
          {item.keywords.map(keyword => (
            <Text key={keyword}>{keyword}</Text>
          ))}
        </Box>
      </Box>
    )
  })

  return (
    <Box margin="none" pad="none" border="all">
      <ResumeSection>
        <Heading
          level={3}
          size="medium"
          align="center"
          alignSelf="center"
          textAlign="center"
          margin="none"
          pad="none"
        >
          <Text>Skills</Text>
        </Heading>
        <Box direction="column">{renderItems}</Box>
      </ResumeSection>
    </Box>
  )
}

Skills.defaultProps = {
  skills: [],
}

export default Skills
