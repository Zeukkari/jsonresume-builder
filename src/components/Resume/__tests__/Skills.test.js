import React from 'react';
import renderer from 'react-test-renderer';

import Skills from '../Skills';

import defaultResume from '../../../util/defaultResume';


test('renders skills section', () => {
  const data = defaultResume.skills;
  const component = renderer.create(
    <Skills skills={data} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});