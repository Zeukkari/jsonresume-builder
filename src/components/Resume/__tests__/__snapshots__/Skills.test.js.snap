// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders awards section 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 IRSNj"
>
  <div
    className="StyledBox-sc-13pk1d4-0 IRSNj"
  >
    <p
      className="StyledParagraph-tbetod-0 kxFtnQ"
    >
      name:
    </p>
    <p
      className="StyledParagraph-tbetod-0 kxFtnQ"
    >
      level:
      Master
    </p>
    <ul>
      <li>
        HTML
      </li>
      <li>
        CSS
      </li>
      <li>
        Javascript
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders skills section 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 IRSNj"
>
  <div
    className="StyledBox-sc-13pk1d4-0 IRSNj"
  >
    <p
      className="StyledParagraph-tbetod-0 kxFtnQ"
    >
      name:
    </p>
    <p
      className="StyledParagraph-tbetod-0 kxFtnQ"
    >
      level:
      Master
    </p>
    <ul>
      <li>
        HTML
      </li>
      <li>
        CSS
      </li>
      <li>
        Javascript
      </li>
    </ul>
  </div>
</div>
`;
